crates/router/src/core/mandate.rs

```diff
@@ -62,6 +62,40 @@ pub async fn revoke_mandate(
     ))
 }
 
+#[instrument(skip(state))]
+pub async fn update_connector_mandate_id(
```

Comment:
We should use update trackers(trait)/post update trackers(trait) to update the models and get trackers(trait) to get the models. We can have a reusable function which can be called in trait implementation.